src/js/_enqueues/admin/post.js

```diff
@@ -557,7 +557,20 @@ jQuery( function($) {
 			data: data,
 			url: ajaxurl
 		});
-	});
+	}).on(
+		'pageshow.edit-post',
+		/**
+		 * Makes sure post lock is re-established when editor is restored from bfcache.
+		 *
+		 * @param {jQuery.Event} event
+		 * @param {PageTransitionEvent} event.originalEvent
+		 */
+		function ( event ) {
+			if ( event.originalEvent.persisted ) {
+				window.location.reload();
```

Comment:
In practice this code will never fire because this code is loaded on the Classic Editor and TinyMCE disables bfcache due to an `unload` event handler.

Reply:
Nevertheless, _if_ that offending TinyMCE code is commented-out _and_ `Cache-Control: no-store` is disabled (e.g. via [plugin](https://gist.github.com/westonruter/8c19d87a80a36e8f24db910750162628)), then the experience is not great when you leave with a dirty editor state: you get the "Are you sure?" dialog when you leave and then when you go back, to get it again when it attempts to reload: 

[Screen recording 2023-10-02 16.03.53.webm](https://github.com/WordPress/wordpress-develop/assets/134745/9c8bc069-38e3-4c5e-8d6f-ead0112773ae)

Reply:
In the end, this code is not needed because upon `pageshow`, heartbeat will `resume` and it will start sending its `wp-refresh-post-lock` requests.

Reply:
So this `pageshow` event is removed in 329ebfbb2932dbddb69a66d71231131c60d4b8d3, and then heartbeat then connects immediately upon `pageshow` if restoring from bfcache in 00772ee2907d2b736c1fbbb36871f641fe4bbe7d. 

Much better experience:

[Screen recording 2023-10-02 16.19.16.webm](https://github.com/WordPress/wordpress-develop/assets/134745/b9c0b1c8-6332-4677-b471-a3e9dd1042a6)